datanode/api/trading_data_v2.go

```diff
@@ -543,8 +543,16 @@ func (t *tradingDataServiceV2) ListCandleData(ctx context.Context, req *v2.ListC
 		return nil, errors.New("sql candle service not available")
 	}
 
-	from := vegatime.UnixNano(req.FromTimestamp)
-	to := vegatime.UnixNano(req.ToTimestamp)
+	var from, to *time.Time
+	if req.FromTimestamp != 0 {
+		fromTimestamp := vegatime.UnixNano(req.FromTimestamp)
+		from = &fromTimestamp
```

Comment:
Nitpick: use the ptr lib (lib/ptr)
```suggestion
		from = ptr.From(vegatime.UnixNano(req.FromTimestamp))
```

Reply:
Thanks, I wasn't aware of that package/function. I've updated with this commit https://github.com/vegaprotocol/vega/pull/6267/commits/d7e76cc8032cd7103ffa7dc7a1bd6b267c2392e9